### PR TITLE
[Enhancement] Email birthday reminder

### PR DESCRIPTION
**Pull request by Mangesh-Khairnar**
Birthday reminder were not getting sent through email initially, since, recipients were not correctly selected. Also, the function was designed to send multiple emails, when multiple employees had their birthdays on the same day.
Now, Single email is sent for multiple birthdays. This includes sending a personalise email for the situation in which more than one person has a birthday on the same day. These personalise emails are designed in such a way that an employee celebrating a birthday on a particular day wouldn't get a email reminder for their own birthday but they would get a reminder for the birthday of their colleagues.

### CODE REVIEW

File: erpnext/hr/doctype/employee/employee.py [hunk 1]
```diff
@@ -223,24 +223,21 @@ def send_birthday_reminders():
...
+			frappe.sendmail(recipients=filter(lambda u: u not in (e.company_email, e.personal_email, e.user_id), employee_emails),
 				subject=_("Birthday Reminder for {0}").format(e.employee_name),
-				message=_("""Today is {0}'s birthday!""").format(e.employee_name),
+				message=_("""Today is {0}'s birthday! \U0001F601""").format(e.employee_name),
 				reply_to=e.company_email or e.personal_email or e.user_id)
 
 def get_employees_who_are_born_today():
 	"""Get Employee properties whose birthday is today."""
-	return frappe.db.sql("""select name, personal_email, company_email, user_id, employee_name
+	return frappe.db.sql("""select name,personal_email, company, company_email, user_id, employee_name
```
Comment (netchampfaris) [inline erpnext/hr/doctype/employee/employee.py]: Use ORM instead of raw sql

File: erpnext/hr/doctype/employee/employee.py [hunk 2]
```diff
@@ -223,24 +223,21 @@ def send_birthday_reminders():
...
 
 		for e in birthdays:
-			frappe.sendmail(recipients=filter(lambda u: u not in (e.company_email, e.personal_email, e.user_id), users),
+			employee_emails = frappe.db.get_all('Employee', filters={'company': e.company, 'status': 'Active'}, fields=['company_email'])
+			employee_emails = [email["company_email"] for email in employee_emails]
+			frappe.sendmail(recipients=filter(lambda u: u not in (e.company_email, e.personal_email, e.user_id), employee_emails),
 				subject=_("Birthday Reminder for {0}").format(e.employee_name),
-				message=_("""Today is {0}'s birthday!""").format(e.employee_name),
+				message=_("""Today is {0}'s birthday! \U0001F601""").format(e.employee_name),
 				reply_to=e.company_email or e.personal_email or e.user_id)
```
Comment (netchampfaris) [inline erpnext/hr/doctype/employee/employee.py]: add header

File: erpnext/hr/doctype/employee/employee.py [hunk 3]
```diff
@@ -223,24 +223,21 @@ def send_birthday_reminders():
...
 
 	if birthdays:
-		if not users:
-			users = [u.email_id or u.name for u in get_enabled_system_users()]
 
 		for e in birthdays:
-			frappe.sendmail(recipients=filter(lambda u: u not in (e.company_email, e.personal_email, e.user_id), users),
+			employee_emails = frappe.db.get_all('Employee', filters={'company': e.company, 'status': 'Active'}, fields=['company_email'])
+			employee_emails = [email["company_email"] for email in employee_emails]
+			frappe.sendmail(recipients=filter(lambda u: u not in (e.company_email, e.personal_email, e.user_id), employee_emails),
```
Comment (netchampfaris) [inline erpnext/hr/doctype/employee/employee.py]: move this logic into a function

**Review by netchampfaris — CHANGES_REQUESTED** [review]

**Comment by rmehta** [comment]
👍  for first PR

**Comment by rmehta** [comment]
Please add screenshots of the mail that will be sent.

**Pull request closed without merging**